MyFirstSlideshow/app/src/main/java/com/charles/myfirstslideshow/MainActivity.java

```diff
@@ -10,12 +26,141 @@
  *  This code is for demonstration purposes only. It should not be assumed to represent best
  *  practices or patterns employed at Yoti.
  */
+public class MainActivity extends AppCompatActivity implements Serializable {
+
+    // Images under Creative Commons
+    // Images attributed to (in order) https://www.flickr.com/photos/_torne/
+    // https://www.flickr.com/photos/chrisyarzab/
+    // https://www.reddit.com/user/lalien42/
+    // http://www.kapstadt.de/
+
+    public static List images = Arrays.asList(new String[]{"https://c1.staticflickr.com/6/5615/15570202337_0e64f5046e_k.jpg",
+            "https://c1.staticflickr.com/4/3169/2846544061_cb7c04b46f_b.jpg",
+            "https://i.redd.it/d8q1wkgu1awy.jpg",
+            "http://www.kapstadt.de/webcam.jpg"});
 
-public class MainActivity extends AppCompatActivity {
+    static int i = 0;
+    static int j = 0;
+
+    volatile WeakReference<ImageView> image1;
+    volatile WeakReference<ImageView> image2;
```

Comment:
If you are using "volatile" keyword, you should definitely explain why.  It looks like there is no good reason for it on the first glance. Explain it or remove the "volatile" keyword.